pkg/server/container_create.go

```diff
@@ -147,6 +148,9 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate container %q spec: %v", id, err)
 	}
+	spec.Annotations[annotations.SandboxID] = sandboxID
+	spec.Annotations[annotations.SandboxName] = name
```

Comment:
1. Can we do this in `generateContainerSpec`?
2. `name` is the name of the container, right? Not the sandbox name. What name do you need here? From the annotation you applied on sandbox, it seems that you require the pod name?

Reply:
That is true this is pod name, I will provide the sandbox struct  to generateContainerSpec and populate with the right value.